src/pages/costModels/costModel/updateDistributionDialog.tsx

```diff
@@ -0,0 +1,126 @@
+import { costModelsActions, costModelsSelectors } from 'store/costModels';
+
+interface Props extends WithTranslation {
+  isLoading: boolean;
+  onClose: typeof costModelsActions.setCostModelDialog;
+  updateCostModel: typeof costModelsActions.updateCostModel;
+  error: string;
+  current: CostModel;
+}
+
+interface State {
+  distribution: string;
+}
+
+class UpdateDistributionModelBase extends React.Component<Props, State> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      distribution: this.props.current.distribution,
+    };
+  }
+
+  private handleDistributionChange = (_, event) => {
+    const { value } = event.currentTarget;
+    this.setState({ distribution: value });
+  };
+
+  public render() {
+    const { error, current, onClose, updateCostModel, isLoading, t } = this.props;
+    return (
+      <Modal
+        title={t('cost_models_details.distribution_type')}
+        isOpen
+        onClose={() => onClose({ name: 'updateDistribution', isOpen: false })}
+        variant="small"
+        actions={[
+          <Button
+            key="proceed"
+            variant="primary"
+            onClick={() => {
+              const newState = {
+                ...current,
+                source_uuids: current.sources.map(provider => provider.uuid),
+                // will always be OCP
+                source_type: 'OCP',
+                distribution: this.state.distribution,
+              };
+              updateCostModel(current.uuid, newState, 'updateDistribution');
+            }}
+          >
+            {t('save')}
+          </Button>,
+          <Button
+            key="cancel"
+            variant="link"
+            onClick={() => onClose({ name: 'updateDistribution', isOpen: false })}
+            isDisabled={isLoading}
+          >
+            {t('cancel')}
+          </Button>,
+        ]}
+      >
+        <Stack hasGutter>
+          <StackItem>{error && <Alert variant="danger" title={`${error}`} />}</StackItem>
+          <StackItem>
+            <Title headingLevel="h2" size="md">
+              {t('cost_models_details.description_distribution_model')}
+            </Title>
+          </StackItem>
+          <StackItem>
+            <Form>
+              <FormGroup isInline fieldId="cost-distribution" isRequired>
+                <Radio
+                  isChecked={this.state.distribution === 'cpu'}
+                  name="cpuDistribution"
+                  label="CPU"
+                  id="cpuDistribution"
+                  value="cpu"
+                  onChange={this.handleDistributionChange}
+                />
+                <Radio
+                  isChecked={this.state.distribution === 'memory'}
+                  name="memoryDistribution"
+                  label="Memory"
+                  id="memoryDistribution"
+                  value="memory"
+                  onChange={this.handleDistributionChange}
+                />
```

Comment:
We should add aria attributes here. Suspect PatternFly is outputting warnings in the browser console without them?

Reply:
Added